react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log('state:', this.state);
+  }
+
+  render() {
+    return (
+        <form onSubmit={this.handleSubmit}>
+          <label>
+            Email:
+            <input type="text" name="email" onChange={this.handleChange} />
```

Comment:
Be sure to also pass a `value` prop to your input so that it is fully "controlled".